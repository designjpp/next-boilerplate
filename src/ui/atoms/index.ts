export * from './button';
export * from './mouse';
export * from './scrollbar';
export * from './wow-title';
