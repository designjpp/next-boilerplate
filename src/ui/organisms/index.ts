// Put your organisms here
export const Organism = () => null;
