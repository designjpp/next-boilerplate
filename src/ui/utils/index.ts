export * from './media-queries';
export * from './tailwind-config';
