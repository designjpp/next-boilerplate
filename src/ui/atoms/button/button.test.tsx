import { render } from '@testing-library/react';
import { axe } from 'jest-axe';
import { create } from 'react-test-renderer';

import { Button } from 'ui/atoms';

describe('<Button>', () => {
	it("should match is's snapshot", () => {
		const component = create(<Button />).toJSON();
		expect(component).toMatchSnapshot();
	});

	it("should match is's snapshot for loading state", () => {
		const component = create(<Button loading={true} />).toJSON();
		expect(component).toMatchSnapshot('loading');
	});

	it('should be accessible', async () => {
		const { container } = render(<Button />);
		const results = await axe(container);
		expect(results).toHaveNoViolations();
	});
});
