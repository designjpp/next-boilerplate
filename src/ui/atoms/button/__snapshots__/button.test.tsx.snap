// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GButton> should match is's snapshot 1`] = `
<button
  aria-disabled={false}
  aria-label="button"
  className=" medium "
  data-testid="button"
  disabled={false}
  role="button"
  type="button"
>
  <span
    aria-hidden={true}
    role="img"
  />
  <span
    aria-hidden={true}
    role="img"
  />
</button>
`;
